### ISSUE TITLE
Support ReceiveAndDelete mode

### ISSUE THREAD

**Issue by jeffhollan** [Azure/azure-functions-servicebus-extension#45]
Customer (from CSE) requested ability to change the trigger mode to receive and delete.  They understand messages may be lost but want the higher throughput that pattern can create.

**Comment by markmcgookin** [comment]
I too would like this, or the ability to complete a message manually... we have a few one and done jobs that might take a long time to run (but might not) and in debugging they seem to lose their lock and the job starts a second time, these jobs can't be run multiple times (they email customers etc..) so I would rather that it bombed out than to retry.

**Comment by FreyMo** [comment]
Is there any news on this? Is it planned to add something like a `ReceiveMode` property to Azure Function apps with the values `ReceiveAndDelete` and `PeekLock` or will it always be `PeekLock` for Service Bus Triggers in Azure Functions? Thanks in advance.

**Comment by jeffhollan** [comment]
Probably not always, but this isn't scheduled for work in the next few sprints currently.  More feedback is always welcome though to help make that call. I suspect the reason you are interested in ReceiveAndDelete for the potential throughput improvements?  We could work with @fabiocav but this may also be a feature that could be contributed as well if any subscribed to issue are interested in opening a PR

**Comment by FreyMo** [comment]
Yes, it is about the throughput. Also, it feels more 'complete' to have the same features as the Service Bus SDK available to Azure Functions. I can however understand the default to 'PeekLock' as it offers the most flexibility and the performance is sufficient for most scenarios.

**Comment by svermaak** [comment]
Also keen to have ReceiveAndDelete

**Comment by dcbrown16** [comment]
We have received a support request asking for this capability also. Please reach out to me at v-dibr internally if you need more details. The scenario is that they are encountering double sending with the PeekLock behavior.

**Comment by Bit-Shifts** [comment]
I am also keen on this feature. After receiving a message from either queue or topic I'd like to start a long running process from message data. I can't now as the Lock will be lost if it runs longer then 5 minutes.
